src/components/ColorPicker/commons/saturation/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { useRef, useEffect, useContext } from 'react';
+import { UP_KEY, DOWN_KEY, RIGHT_KEY, LEFT_KEY } from '../../../../libs/constants';
+import {
+    hsvToRgb,
+    rgbToRgba,
+    recomposeColor,
+    decomposeColor,
+    rgbaToHex,
+} from '../../../../styles/helpers/color';
+import { StyledColor, StyledCircle } from './styled';
+import { ColorPickerContext } from '../../context';
+import { calculateSaturation, calculateBright } from './helpers';
+
+const Saturation = React.forwardRef((_props, ref) => {
+    const { rgba, hsv, tabIndex, onChange } = useContext(ColorPickerContext);
+    const containerRef = useRef();
+    const [h, s, v] = hsv;
+    const a = rgba[3];
+
+    const change = ({ saturation, bright }) => {
+        if (saturation >= 0) {
+            hsv[1] = saturation;
+        }
+        if (bright >= 0) {
+            hsv[2] = bright;
+        }
+        const rgbaColor = rgbToRgba(hsvToRgb(recomposeColor({ type: 'hsv', values: hsv })), a);
+
+        onChange({
+            hex: `#${rgbaToHex(rgbaColor)}`,
+            rgba: decomposeColor(rgbaColor).values,
+            hsv,
+        });
+    };
+
+    const handleChange = event => {
+        const rect = containerRef.current.getBoundingClientRect();
+        const saturation = calculateSaturation(event, rect);
+        const bright = calculateBright(event, rect);
+
+        change({ saturation, bright });
+    };
+
+    const unbindEventListeners = () => {
+        window.removeEventListener('mousemove', handleChange);
+        window.removeEventListener('mouseup', unbindEventListeners);
+    };
+
+    const handleMouseDown = event => {
+        handleChange(event);
+        window.addEventListener('mousemove', handleChange);
+        window.addEventListener('mouseup', unbindEventListeners);
```

Comment:
can we avoid setting listener in window and instead attach it to specific element to reduce its scope?

Reply:
I need to keep listening for the move event when it is outside the component so that the behavior is like the native component